src/test/java/io/appium/java_client/pagefactory_tests/TimeoutTest.java

```diff
@@ -0,0 +1,132 @@
+import static java.lang.Math.abs;
+import static java.lang.String.format;
+import static java.lang.System.currentTimeMillis;
+import static java.lang.System.setProperty;
+import static java.util.concurrent.TimeUnit.MICROSECONDS;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.hamcrest.Matchers.lessThanOrEqualTo;
+import static org.junit.Assert.assertThat;
+import static org.openqa.selenium.support.PageFactory.initElements;
+
+import io.appium.java_client.pagefactory.AppiumFieldDecorator;
+import io.appium.java_client.pagefactory.TimeOutDuration;
+import io.appium.java_client.pagefactory.WithTimeout;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.chrome.ChromeDriverService;
+import org.openqa.selenium.support.FindAll;
+import org.openqa.selenium.support.FindBy;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+public class TimeoutTest {
+
+    private static final long ACCEPTABLE_TIME_DIFF = 1500;
+    private static final String MESSAGE = "Check difference from the expected waiting duration %s %s";
+
+    private WebDriver driver;
+
+    @FindAll({
+            @FindBy(className = "ClassWhichDoesNotExist"),
+            @FindBy(className = "OneAnotherClassWhichDoesNotExist")})
+    private List<WebElement> stubElements;
+
+    @WithTimeout(time = 5, unit = SECONDS)
+    @FindAll({@FindBy(className = "ClassWhichDoesNotExist"),
+            @FindBy(className = "OneAnotherClassWhichDoesNotExist")})
+    private List<WebElement> stubElements2;
+
+    private TimeOutDuration timeOutDuration;
+
+    private static long getBenchmark(Runnable runnable) {
+        long startMark = currentTimeMillis();
+        runnable.run();
+        long endMark = currentTimeMillis();
+        return endMark - startMark;
+    }
+
+    private static long getExpectedMillis(long value, TimeUnit sourceTimeUnit) {
+        return MILLISECONDS.convert(value, sourceTimeUnit);
+    }
+
+    /**
+     * The setting up.
+     */
+    @Before public void setUp() throws Exception {
+        setProperty(ChromeDriverService.CHROME_DRIVER_EXE_PROPERTY,
+                getChromeDriver().getAbsolutePath());
+        driver = new ChromeDriver();
+        timeOutDuration = new TimeOutDuration(DEFAULT_TIMEOUT, DEFAULT_TIMEUNIT);
+        initElements(new AppiumFieldDecorator(driver, timeOutDuration), this);
+    }
+
+    /**
+     * finishing.
+     */
+    @After public void tearDown() throws Exception {
+        driver.quit();
+    }
+
+    @Test public void defaultTimeOutTest() {
+        assertThat(format(MESSAGE, DEFAULT_TIMEOUT, DEFAULT_TIMEUNIT),
+                abs(getExpectedMillis(DEFAULT_TIMEOUT, DEFAULT_TIMEUNIT) - getBenchmark(() -> stubElements.size())),
```

Comment:
I assume the whole formula can be extracted into a separate method, for example:
```java
private static long getPerformanceDiff(long expectedMs, Runnable m) {
   returns abs(expectedMs  - getBenchmark(m1))
}
```